File: crates/sem-core/tests/fixtures/scope_test/kotlin/Service.kt
fun createDog(name: String): Dog {
    val dog = Dog(name)
    if (!dog.validate()) {
        throw IllegalArgumentException("invalid dog")
    }
    val conn = getConnection()
    conn.execute("INSERT INTO dogs VALUES (?)")
    conn.commit()
    return dog
}

fun createCat(name: String): Cat {
    val cat = Cat(name)
    if (!cat.validate()) {
        throw IllegalArgumentException("invalid cat")
    }
    val conn = getConnection()
    conn.execute("INSERT INTO cats VALUES (?)")
    conn.commit()
    return cat
}

fun transferAnimal(animal: Any, shelter: Shelter) {
    val txn = Transaction(getConnection())
    txn.execute("UPDATE animals SET shelter_id = ?")
    shelter.add(animal)
    txn.commit()
}

fun listAnimals(): List<Any> {
    val conn = getConnection()
    return conn.execute("SELECT * FROM animals") as List<Any>
}

File: crates/sem-core/tests/fixtures/scope_test/swift/service.swift
func createDog(name: String) -> Dog {
    let dog = Dog(name: name)
    if !dog.validate() {
        fatalError("invalid dog")
    }
    let conn = getConnection()
    conn.execute(query: "INSERT INTO dogs VALUES (?)")
    conn.commit()
    return dog
}

func createCat(name: String) -> Cat {
    let cat = Cat(name: name)
    if !cat.validate() {
        fatalError("invalid cat")
    }
    let conn = getConnection()
    conn.execute(query: "INSERT INTO cats VALUES (?)")
    conn.commit()
    return cat
}

func transferAnimal(animal: Any, shelter: Shelter) {
    let txn = Transaction(conn: getConnection())
    txn.execute(query: "UPDATE animals SET shelter_id = ?")
    shelter.add(animal: animal)
    txn.commit()
}

func listAnimals() -> [Any] {
    let conn = getConnection()
    return conn.execute(query: "SELECT * FROM animals") as! [Any]
}
